lib/monitor/run.js

```diff
@@ -16,6 +16,27 @@ var psTree = require('pstree.remy');
 var path = require('path');
 var signals = require('./signals');
 
+function waitForOldKids(oldKids, callback) {
+  if (oldKids) {
+    // check if all previous kids have been terminated
+    exec('kill -0  ' + oldKids.join(' '), (error) => {
+      const returnCode = error ? error.code : 0;
+      if (returnCode < 126) { // ignore command not found error
+        const stillRunningKids = oldKids.length - returnCode;
+        if (stillRunningKids > 0) {
+          utils.log.status('still waiting for ' + stillRunningKids +
+            ' child process(es) to finish...');
+          setTimeout(waitForOldKids.bind(this, oldKids, callback), 100);
```

Comment:
A slower polling interval might be nice for processes that aren't actually planning to exit.